server/queue/queue-api/src/test/java/org/apache/james/queue/api/MailQueueMetricContract.java

```diff
@@ -66,59 +75,67 @@ default void deQueueMail(Integer times) {
+        assertThat(testSystem.getMetricFactory().countForPrefixName(ENQUEUED_METRIC_NAME_PREFIX))
+            .hasSize(1)
+            .containsExactly(2);
     }
 
     @Test
-    default void enqueueShouldNotTouchDequeueMetric(MailQueueMetricExtension.MailQueueMetricTestSystem testSystem) throws Exception {
+    default void enqueueShouldNotTouchDequeueMetric(MailQueueMetricExtension.MailQueueMetricTestSystem testSystem) {
         enQueueMail(2);
 
-        verify(testSystem.getSpyEnqueuedMailsMetric(), times(2)).increment();
-        Mockito.verifyNoMoreInteractions(testSystem.getSpyDequeuedMailsMetric());
+        assertThat(testSystem.getMetricFactory().countForPrefixName(ENQUEUED_METRIC_NAME_PREFIX))
+            .hasSize(1)
+            .containsExactly(2);
+        assertThat(testSystem.getMetricFactory().countForPrefixName(DEQUEUED_METRIC_NAME_PREFIX))
+            .hasSize(0);
     }
 
     @Test
-    default void dequeueShouldIncreaseDequeueMetric(MailQueueMetricExtension.MailQueueMetricTestSystem testSystem) throws Exception {
+    default void dequeueShouldIncreaseDequeueMetric(MailQueueMetricExtension.MailQueueMetricTestSystem testSystem) {
         enQueueMail(2);
         deQueueMail(2);
 
-        verify(testSystem.getSpyDequeuedMailsMetric(), times(2)).increment();
-        Mockito.verifyNoMoreInteractions(testSystem.getSpyDequeuedMailsMetric());
+        assertThat(testSystem.getMetricFactory().countForPrefixName(DEQUEUED_METRIC_NAME_PREFIX))
+            .hasSize(1)
+            .containsExactly(2);
     }
 
     @Test
-    default void dequeueShouldNotTouchEnqueueMetric(MailQueueMetricExtension.MailQueueMetricTestSystem testSystem) throws Exception {
+    default void dequeueShouldNotTouchEnqueueMetric(MailQueueMetricExtension.MailQueueMetricTestSystem testSystem) {
         enQueueMail(2);
         deQueueMail(2);
 
-        verify(testSystem.getSpyDequeuedMailsMetric(), times(2)).increment();
-        Mockito.verifyNoMoreInteractions(testSystem.getSpyDequeuedMailsMetric());
-
-        verify(testSystem.getSpyEnqueuedMailsMetric(), times(2)).increment();
-        Mockito.verifyNoMoreInteractions(testSystem.getSpyEnqueuedMailsMetric());
+        assertThat(testSystem.getMetricFactory().countForPrefixName(DEQUEUED_METRIC_NAME_PREFIX))
+            .hasSize(1)
+            .containsExactly(2);
+        assertThat(testSystem.getMetricFactory().countForPrefixName(ENQUEUED_METRIC_NAME_PREFIX))
+            .hasSize(1)
+            .containsExactly(2);
     }
 
     @Test
-    default void enqueueShouldPublishEnqueueTimeMetric(MailQueueMetricExtension.MailQueueMetricTestSystem testSystem) throws Exception {
+    default void enqueueShouldPublishEnqueueTimeMetric(MailQueueMetricExtension.MailQueueMetricTestSystem testSystem) {
         enQueueMail(2);
 
-        verify(testSystem.getSpyEnqueuedMailsTimeMetric(), times(2)).stopAndPublish();
+        WAIT_CONDITION
```

Comment:
enqueue email is synchronous, so when it returns we should not need to await the metric.

Or did I miss something?